File: 2021W13/fienestar/boj_1764.cpp
/*
    듣보잡
    https://www.acmicpc.net/problem/1764
*/
#include <bits/stdc++.h>

using namespace std;

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    size_t N,M;
    cin >> N >> M;

    string name;
    set<string> m;
    while(N--){
        cin >> name;
        m.insert(name);
    }

    set<string> result;
    while(M--){
        cin >> name;
        if(m.find(name) != m.end())
            result.insert(name);
    }

    cout << result.size() << "\n";
    for(auto& name:result)
        cout << name << "\n";
}

File: 2021W29/B0neh3ad/boj_1764.py
'''
    듣보잡
    https://www.acmicpc.net/problem/1764
'''
N, M = map(int, input().split())
L = set()
for i in range(N):
    L.add(input())
S = set()
for i in range(M):
    S.add(input())

L.intersection_update(S)
result = list(L)
result.sort()
print(len(result))
for name in result:
    print(name)